packages/components/src/components/assets/AssetShareIndicator/AssetShareIndicator.tsx

```diff
@@ -1,34 +1,128 @@
+    ({ $size }) => `
+        position: relative;
+        align-items: center;
+        display: flex;
+        justify-content: center;
+        width: ${$size}px;
+        height: ${$size}px;
+        border-radius: 50%;
+        margin: ${spacingsPx.xxs};
+  `,
+);
+
+const rgbToHexColor = (rgbColor: string) => {
+    const [r, g, b] = rgbColor.split('-').map(Number);
+    const hex = [r, g, b].map(x => x.toString(16).padStart(2, '0')).join('');
+
+    return '#' + hex;
+};
+
+const getDominantColor = (imageData: ImageData) => {
+    const colorMap: { [key: string]: number } = {};
+    const numberOfPixelValues = 4;
+    let dominantKey = '';
+    let maxCount = 0;
+
+    for (let i = 0; i < imageData.data.length; i += numberOfPixelValues) {
+        const red = imageData.data[i];
+        const green = imageData.data[i + 1];
+        const blue = imageData.data[i + 2];
+        const alpha = imageData.data[i + 3];
+
+        if (alpha !== 255) {
+            continue; // ignore pixel with transparency
+        }
+
+        const imagePixelKey = `${red}-${green}-${blue}`;
+        colorMap[imagePixelKey] = (colorMap[imagePixelKey] || 0) + 1;
+
+        if (colorMap[imagePixelKey] > maxCount) {
+            dominantKey = imagePixelKey;
+            maxCount = colorMap[imagePixelKey];
+        }
+    }
+
+    return rgbToHexColor(dominantKey);
+};
+
+const loadImageAndProcessColor = (
+    image: HTMLImageElement,
+    size: number,
+    onSetColor: Dispatch<SetStateAction<string | undefined>>,
+) => {
+    const canvas = document.createElement('canvas');
+    const canvasContext = canvas.getContext('2d');
+
+    if (image && canvasContext) {
+        try {
+            canvas.width = size;
+            canvas.height = size;
+            canvasContext.drawImage(image, 0, 0, size, size);
+
+            const imageData = canvasContext.getImageData(0, 0, canvas.width, canvas.height);
+
+            onSetColor(getDominantColor(imageData));
+        } catch (error) {
+            console.error('Error processing image color:', error);
+            onSetColor(undefined);
+        }
+    }
+};
 
 export interface AssetShareIndicatorProps extends CoinLogoProps {
     percentageShare?: number;
+    showProgressCircle?: boolean;
+    firstCharacter?: string;
 }
 
 interface ProgressCircleProps
     extends Pick<AssetShareIndicatorProps, 'symbol' | 'percentageShare' | 'index'> {
+    iconUrl?: string;
     size: number;
 }
 
-const ProgressCircle = ({ symbol, size, percentageShare, index = 0 }: ProgressCircleProps) => {
+const ProgressCircle = ({
+    symbol,
+    iconUrl,
+    size,
+    percentageShare,
+    index = 0,
+}: ProgressCircleProps) => {
     const theme = useTheme();
+    const [dominantColor, setDominantColor] = useState<string | undefined>();
+
+    useEffect(() => {
+        if (!symbol && iconUrl) {
+            const image = new Image();
+
+            if (image) {
+                image.crossOrigin = 'anonymous';
```

Comment:
For me it works fine, but not sure it will work for everybody. Yanas told me he'll try to reach Tali and ask what we can do regarding the server configuration.